### DISCUSSION TITLE
How can we configure Kiali to use a remote Prometheus with mTLS enabled?

### DISCUSSION THREAD

**Discussion by imaskm** [kiali/kiali#4397] [Q&A]
Hi guys, I need to use a remote Prometheus(installed on another cluster) on Kiali but the Prometheus is exposed via mTLS and istio gateway,  I have figured out everything except how to tell kiali to use mTLS for Prometheus APIs.

Looking for some helpful suggestions, thanks in advance!!

**Comment by jmazzitelli** [comment]
I assume this isn't helpful: https://github.com/kiali/kiali-operator/blob/v1.40.1/deploy/kiali/kiali_cr.yaml#L584-L619

Have you tried putting a sidecar on the Kiali pod and have the mTLS traffic go through the sidecar?

**Reply by imaskm** [reply]
Yes, that won't work. I even check how kiali is creating Prometheus client, there is an option to pass CA cert with Auth but not to pass key pair.  I didn't understand your sidecar approach? Isn't Kiali will be calling the Prometheus endpoints?

**Comment by jmazzitelli** [comment]
> Isn't Kiali will be calling the Prometheus endpoints?

Well, I'm thinking how the rest of Istio/service mesh components work. The sidecar approach is supposed to make the mTLS communications an external concern (external to the microservice). The microservice itself can just send http traffic and let the sidecar put the traffic on the mesh over mTLS like any other microservice in the mesh.

So you put a sidecar on Kiali, and assuming Prometheus is also in the mesh, it should "just work."

I'm sure I'm simplifying it, but isn't this how a service mesh is supposed to work? The components don't worry about mTLS - that's the proxy's job?

**Reply by imaskm** [reply]
Actually, the prometheus in on a different cluster and that's why I exposed it publically via mTLS.

**Reply by jmazzitelli** [reply]
> the prometheus in on a different cluster

You can use a Istio [ServiceEntry](https://istio.io/latest/docs/reference/config/networking/service-entry/) to send traffic out.

**Reply by israel-hdez** [reply]
I wonder if for `mTLS` you are referring to [Istio's mTLS feature](https://istio.io/latest/docs/concepts/security/#mutual-tls-authentication), or if you are trying to tell that that your Prometheus is exposed using HTTPS only rather than plain HTTP???

EDIT: Although I see in the main comment that you are mentioning an Istio gateway... So, I guess you are referring to Istio's mTLS feature.

I would like to understand a little more:
* The "Istio gateway" you are mentioning is an ingress gateway, or an egress gateway?
* The remote Prometheus is part of the mesh AND the same mesh of Kiali, or are both services on different Istio meshes?
